### ISSUE TITLE
Opening file from explorer causes trouble when used behind a reverse proxy (2.1698 release)

### ISSUE THREAD

**Issue by LukasMasuch** [cdr/code-server#1294]
- `code-server` version: 2.1698-vsc1.41.1
- OS Version: Ubuntu 18.04
- Extension: [vscode-python](https://github.com/microsoft/vscode-python)(2020.1.58038)

## Description

If the code server is opened without providing a `folder` via URL parameter, everything is working fine. However, in case the `folder` parameter is used and the file is opened via the Explorer, the python extension doesn't seem to be able to fully interact with the (python-) file (underlines are missing, formatter is not working...):

<img width="971" alt="Screenshot at Jan 24 13-50-59" src="https://user-images.githubusercontent.com/2852129/73070808-94837c80-3eb1-11ea-904a-3462396fe03f.png">

However, if the file is opened via `File -> Open`, a new tab is created and everything is working fine:

<img width="974" alt="Screenshot at Jan 24 13-51-22" src="https://user-images.githubusercontent.com/2852129/73070820-98170380-3eb1-11ea-93c3-a450fe4b2cf3.png">

Based on the tab labels, it seems like that opening files via `File -> Open` is using absolute paths while the explorer uses a relative path (which is causing problems with the python extension). This problem does not happen with the older code-server versions (e.g. 2.1692-vsc1.39.2). I also looked into the logs, but wasn't able to find anything suspicious. 

To summarize:

- Older code-server version -> working
- Opening code-server without `folder` parameter -> working
- Opening code-server with `folder` parameter:
	- Open python file with `File -> Open` -> working
	- Open python file from explorer -> not working

**Comment by nhooyr** [comment]
Will test and get back to you soon.

**Comment by nhooyr** [comment]
Not sure what's going on here. I get no underline no matter what. Are there any errors in the console?

**Comment by LukasMasuch** [comment]
Thanks for investigating! The underlines are warning/errors from python lintings and that can depend on the python extension settings within VS code. I think the better way of debugging or reproducing this issue is by checking if opening the file via the explorer (file browser) and, subsequently, opening the same file via `File -> Open` will produce two tabs with the same content (issue) or only one (how it should be). 

I also did a bit more investigation and discovered that the issue actually has nothing specific to do with the python extension itself and will, most likely, cause problems with other functionalities and extensions as well. Also, the issue only appears when the VS Code instance is accessed from a base URL (e.g. via a reverse proxy).

Here is an update an the setups I have tested that are working, and the setup which is causing trouble:
- Older code-server version -> working
- Opening code server without a URL path (e.g. via reverse proxy) -> working
- Opening code-server without `folder` parameter -> working
- Opening code-server with a base path and `folder` parameter:
	- Open a file with `File -> Open` -> working
	- Open a file from explorer -> not working (File opens in a new tab and causes trouble with functionalities that work on the file, e.g. linting...). Here are some errors logged in the `Log (Window)` section:

![image](https://user-images.githubusercontent.com/2852129/73676564-e3e95a00-46b4-11ea-949c-4fef9115098b.png)

![image](https://user-images.githubusercontent.com/2852129/73676990-a89b5b00-46b5-11ea-99dd-7caa315aa885.png)

![image](https://user-images.githubusercontent.com/2852129/73677090-d41e4580-46b5-11ea-9af3-c74acdda37d5.png)

**Comment by LukasMasuch** [comment]
I also renamed the issue since I believe that it is much more related to the usage of code-server behind a reverse proxy, and not explicitly related to the python extension.

**Comment by nhooyr** [comment]
@code-asher

**Comment by orditeck** [comment]
- `code-server` version: 2.1698-vsc1.41.1
- OS Version: Ubuntu 18.04

## Description

When editing JavaScript, I can't format the document without using an odd workaround. 

## Steps to Reproduce

1. Open a JS file
2. Open command palette -> Format document
3. Get message There is no formatter for 'javascript' files installed.

In the "Problems" tab, I can see that there is an ESLint error:
![Screen Shot 2020-02-08 at 10 23 17 PM](https://user-images.githubusercontent.com/2186535/74095632-a5dba400-4ac1-11ea-90d0-27f5b39bc54a.png)

But the error isn't displayed in the editor:
![Screen Shot 2020-02-08 at 10 23 55 PM](https://user-images.githubusercontent.com/2186535/74095636-b9870a80-4ac1-11ea-9be7-004e11fb7b81.png)

Then, if I click on the file shown in the error:
![Screen Shot 2020-02-08 at 10 23 17 PM](https://user-images.githubusercontent.com/2186535/74095632-a5dba400-4ac1-11ea-90d0-27f5b39bc54a.png)

It re-opens the exact same file, but this time, with ESLint working:
![image](https://user-images.githubusercontent.com/2186535/74095647-db808d00-4ac1-11ea-8e8e-bd89d126af35.png)

And I now see the formatter(s):
![image](https://user-images.githubusercontent.com/2186535/74095663-04088700-4ac2-11ea-8ef8-c56539768e92.png)

While in the same file (that I opened from the Explorer), I don't even have "Format document with"):
![image](https://user-images.githubusercontent.com/2186535/74095667-171b5700-4ac2-11ea-819b-f090492caa62.png)

On the other file (opened from the "Problems" tab), I do have the option...
![image](https://user-images.githubusercontent.com/2186535/74095681-34e8bc00-4ac2-11ea-896b-2be83e6bfe2b.png)

Reverting back to `2.1692-vsc1.39.2` solves the issue. I must say, I absolutely love this project and I would never complaint about something free that is made by awesome people on their free time. This is the best thing I've used in a while, I've been using this software ~40h/week since October 2019. That being said, the previous issues are somewhat worrisome and the previous release (removed) didn't seem to have been tested at all. This release is also having a huge bug (this one) making it barely usable (code formatting is an important feature). I'm saying barely because it still works when using the workaround of opening the file from the "Problems" tab.

From the look of a few comments in other issues, some pretty important decisions seem to have been taken rashly without being discussed between you (the maintainers) and without asking your users base. 

This is purely a comment based on what I've felt from other comments and the fact that I'm worried only because I'm very much attached to this project and the freedom it provides me in my workflow. I just open any computer and I can code.

I hope you guys can work this out! You're doing an amazing work and know that it's very appreciated.

**Comment by orditeck** [comment]
> I also renamed the issue since I believe that it is much more related to the usage of code-server behind a reverse proxy, and not explicitly related to the python extension.

I also use a reverse proxy.
I just tried without my proxy. 
It works.

Reverse proxy issue confirmed.

`2.1692-vsc1.39.2` works with the reverse proxy though.

**Comment by orditeck** [comment]
Hmm...

I've been doing some test.
**It does work using my reverse proxy.**

It's when using the **`?folder=`** argument in URL that it doesn't work, reverse proxy or not. @LukasMasuch, could you check it out?

My guess is that it has something to do with the workspaces. I've been experiencing something else that I meant to look into: with `2.1692` (`vsc1.39.2`), `?folder=` always worked. Since `2.1698` (`vsc1.41.1`), sometimes the `?folder=` argument isn't considered at all, it just loads the last opened project no matter what.

**Comment by code-asher** [comment]
I haven't had any luck replicating this so far on 2.1698. The folder
parameter always seems to work for me as expected. I've been trying
various combinations of opening workspaces and folders and changing the
last visited property in the ~/.local/share/code-server/coder.json file.

There was a bug in the logic that saved the last visited folder or
workspace which I fixed some time ago but isn't in a release yet. But
I'm not sure how or if that would cause this issue. Still, maybe it's
worth trying to delete that file and see if it does anything.

I'll keep poking around for a bit and see if I can find anything. If
anyone has any ideas for what I can do to replicate the issue please let
me know. Thanks!

**Comment by LukasMasuch** [comment]
@orditeck @code-asher Thanks for looking into this issue! I put some more time into figuring out an easy way to replicate the issue. And it actually seems like a general problem within code-server and not related to the reverse proxy:

1. Start code-server via the latest Docker image:
```
docker run -it -p 8123:8080 codercom/code-server --auth none /home/coder/project
```
2. Visit: `http://localhost:8123/?folder=/home/coder/project`
3. Create a new file within the opened project/folder inside VS Code, e.g.: `Explorer -> New File -> test.txt` -> at that point you should have one open editor tab in VS Code with the created file.
![image](https://user-images.githubusercontent.com/2852129/74279361-0f0a2400-4d1b-11ea-841a-b75d9a03a03d.png)
4. Open the same file via the menu: `Menu -> File -> Open -> test.txt` -> at that point another editor tab with the same file should be open:
![image](https://user-images.githubusercontent.com/2852129/74279517-585a7380-4d1b-11ea-849a-8bc8b84ce922.png)
![image](https://user-images.githubusercontent.com/2852129/74278283-437ce080-4d19-11ea-8785-5a4d836c7288.png)
And this is the issue. The expected behavior would be that no new editor tab is opened for the same (already opened) file. Having two tabs with the same file causes some errors within the VS Code logs. Also, the tab opened via the explorer (the tab which was opened in step 3) does seem to cause various problems with vs-code extensions (e.g., python -> formatting, linting...). As stated above, this only happens when the folder parameter is used. Older versions also do not seem to have this issue.

**Comment by code-asher** [comment]
Aha! Thanks for those steps, I was able to repro. Should be able to
track it down from here.

**Comment by code-asher** [comment]
It's apparently related to #1351. So `http://localhost:8080/?folder=vscode-remote%3A%2F%2Flocalhost%3A8080%2Fhome%2Fcoder%2Fproject` works but `http://localhost:8008/?folder=/home/coder/project` doesn't. That explains why I couldn't replicate earlier since I was using the folder picker which uses the former.

So I'll be looking through how they use it and try making it work for both the new format and the old format.

**Comment by LukasMasuch** [comment]
@code-asher Thanks for fixing the issue 👍I really appreciate this work! I will test it once a binary is available.
